packages/core/src/logs/eventMapper.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+import { InternalLog } from '../InternalLog';
+import { SdkVerbosity } from '../SdkVerbosity';
+import { DdSdk } from '../foundation';
+import type { Attributes } from '../sdk/AttributesSingleton/types';
+import type { UserInfo } from '../sdk/UserInfoSingleton/types';
+
+import type { LogEvent, LogEventMapper, LogStatus, RawLog } from './types';
+import { deepClone } from './utils/deepClone';
+
+export const applyLogEventMapper = (
+    logEventMapper: LogEventMapper,
+    log: LogEvent
+): LogEvent | null => {
+    const originalLog = deepClone(log);
+
+    try {
+        const mappedEvent = logEventMapper(log);
+
+        if (!mappedEvent) {
+            return null;
+        }
+
+        return {
+            message: mappedEvent.message,
+            context: mappedEvent.context,
+            status: originalLog.status,
+            userInfo: originalLog.userInfo,
+            attributes: originalLog.attributes
+        };
+    } catch (error) {
+        InternalLog.log(
+            `The log event mapper crashed when mapping log ${JSON.stringify(
+                originalLog
+            )}: ${error}`,
+            SdkVerbosity.WARN
+        );
+        DdSdk.telemetryDebug('Error while running the log event mapper');
```

Comment:
Do we want to potentially separate this out? Currently, if the logEventMapper provided by the user crashes throws, we catch it and send it to our telemetry. Instead, should we only our code in one try / catch to send to telemetry, and a a second try / catch for just the user code?

Reply:
I'm not sure I understand this comment, do you mean that we should only wrap the `const mappedEvent = logEventMapper(log)` in the try/catch block to avoid logging other errors?

Reply:
If I understood correctly, Jeff suggest having a first try/catch around the `const mappedEvent = logEventMapper(log)` line, that would only warn the customer they did something wrong in their mapper implementation, and another try/catch around the rest if needed that would generate a telemetry event

Reply:
So indeed the lines after the mapper invocation can also crash. For instance if the mapper returns a string (if the user thought they should return the log message), it will break when trying to access its `message` property.

In that case:
- I would like to have a telemetry log so we can warn this org if we see this happening for a long time
- I would also like to warn the user since they might not suspect that their log wouldn't change

Now for the telemetry log it can be the same one, maybe we can improve the user warning to differentiate both cases (mapper crash and crash due to improper mapper return value).
Would that be ok for you?

Reply:
Let's start with telemetry enabled, maybe we can remove it if we figure it's triggered too often.